From e12c1a8da19b373aee46f7352e60511ac1a2fc16 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Thu, 1 Mar 2012 17:45:30 +0100
Subject: [PATCH] dracut-functions.sh:install_kmod_with_fw() delay
 .kernelmodseen

first check for omit, then mark the kernel module as seen

when we temporarily omit_drivers, we don't want to mark them as seen.

example: nfs.ko module in kernel-modules, but the nfs module
should be able to load it later on.
---
 dracut-functions.sh |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/dracut-functions.sh b/dracut-functions.sh
index c7337e9..20fc6e1 100755
--- a/dracut-functions.sh
+++ b/dracut-functions.sh
@@ -980,8 +980,6 @@ install_kmod_with_fw() {
 
     [[ -e "$initdir/.kernelmodseen/${1##*/}" ]] && return 0
 
-    > "$initdir/.kernelmodseen/${1##*/}"
-
     if [[ $omit_drivers ]]; then
         local _kmod=${1##*/}
         _kmod=${_kmod%.ko}
@@ -996,6 +994,8 @@ install_kmod_with_fw() {
         fi
     fi
 
+    > "$initdir/.kernelmodseen/${1##*/}"
+
     inst_simple "$1" "/lib/modules/$kernel/${1##*/lib/modules/$kernel/}" \
         || return $?
 
